mono/metadata/metadata.c

```diff
@@ -5154,13 +5150,8 @@ mono_type_stack_size_internal (MonoType *t, int *align, gboolean allow_open)
 {
 	int tmp;
 	MonoTypeEnum simple_type;
-#if TARGET_SIZEOF_VOID_P == SIZEOF_REGISTER
-	int stack_slot_size = MONO_ABI_SIZEOF (gpointer);
-	int stack_slot_align = MONO_ABI_ALIGNOF (gpointer);
-#elif TARGET_SIZEOF_VOID_P < SIZEOF_REGISTER
-	int stack_slot_size = SIZEOF_REGISTER;
-	int stack_slot_align = SIZEOF_REGISTER;
-#endif
+	int stack_slot_size = TARGET_SIZEOF_VOID_P;
+	int stack_slot_align = TARGET_SIZEOF_VOID_P;
 
```

Comment:
Not sure if this will work with ilp32.
Also, the definition of TARGET_SIZEOF_VOID_P might need to be fixed (i.e. target_sizeof_void_p is not always set):
if test "x$target_sizeof_void_p" != "x"; then
   AC_DEFINE_UNQUOTED(TARGET_SIZEOF_VOID_P,${target_sizeof_void_p},[wordsize of target])
   # This is true for all the current cross-compiling targets
   AC_DEFINE(SIZEOF_REGISTER,TARGET_SIZEOF_VOID_P,[size of target machine integer registers])
else
   AC_DEFINE(TARGET_SIZEOF_VOID_P,SIZEOF_VOID_P,[wordsize of target])

Reply:
I'm still confused about ilp32.
All the sizes and alignments should be just like 32bit, I'd expect.
I also suspect it would only be implemented for bitcode only, and that many details would therefore be hidden. Such as register size.
If it isn't bitcode only, then there'd  be details as to if a 32bit parameter occupies an entire register or if a register can hold two values.

Reply:
In either case, I still think there should be no host sensitivity here.
With the caveat that the host might manipulate some target data, on the compile-time evaluation stack, and maybe there are size/alignment factors there.